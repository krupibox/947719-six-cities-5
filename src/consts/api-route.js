export const APIRoute = {
  HOTELS: `/hotels`,
  REVIEWS: `/comments`,
  FAVORITE: `/favorite`,
  LOGIN: `/login`,
};
