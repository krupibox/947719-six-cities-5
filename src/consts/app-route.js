const AppRoute = {
  LOGIN: `/login`,
  RESULT: `/result`,
  ROOT: `/`,
};

export default AppRoute;
