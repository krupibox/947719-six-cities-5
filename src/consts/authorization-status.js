const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export default AuthorizationStatus;
