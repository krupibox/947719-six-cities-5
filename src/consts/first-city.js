export const FIRST_CITY = 0;

