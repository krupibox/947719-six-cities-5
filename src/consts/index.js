export {AuthorizationStatus} from './authorization-status';
export {APIRoute} from './api-route';
export {AppRoute} from './app-route';
