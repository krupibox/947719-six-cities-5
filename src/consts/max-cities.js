const MAX_CITIES = 6;

export default MAX_CITIES;
