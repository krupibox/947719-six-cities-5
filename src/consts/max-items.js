export const MAX_ITEMS = 6;

