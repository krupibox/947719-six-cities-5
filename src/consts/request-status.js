export const RequestStatus = {
  PENDING: `PENDING`,
  SUCCESS: `SUCCESS`,
  FAILURE: `FAILURE`,
  INITIAL: `INITIAL`
};
