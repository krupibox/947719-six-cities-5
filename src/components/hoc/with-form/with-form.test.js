import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import withForm from './with-form';
import {RequestStatus} from '../../../consts/request-status';

const mockStore = configureStore([]);

const store = mockStore({
  REQUEST: {
    status: RequestStatus.SUCCESS,
  }
});

describe(`<withForm/>: `, () => {

  const MockComponent = () => (
    <div/>
  );

  const MockWrapper = withForm(MockComponent);

  it(`should render correctly`, () => {
    const wrapper = renderer
      .create(
          <Provider store={store}>
            <MockWrapper
              offerId={`1`}
            />
          </Provider>)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
