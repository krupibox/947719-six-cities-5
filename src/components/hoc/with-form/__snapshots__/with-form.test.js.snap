// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<withForm/>:  should render correctly 1`] = `<div />`;
